mlir/tools/miopen-gen/miopen-gen.cpp

```diff
@@ -1145,6 +1145,34 @@ const char *getTypeStr(const mlir::Type &type) {
   return "na";
 }
 
+static mlir::Value convertBF16ToFloat(OpBuilder bt0, mlir::Value loadOp) {
+  // HACK for bf16 since BF16 isn't supported by x86 isel
+  auto loc = bt0.getUnknownLoc();
+  mlir::Type i16 = bt0.getIntegerType(16);
+  mlir::Type i32 = bt0.getI32Type();
+  loadOp = bt0.create<arith::BitcastOp>(loc, i16, loadOp);
+  loadOp = bt0.create<arith::ExtUIOp>(loc, i32, loadOp);
+  auto cst16Op = bt0.create<arith::ConstantIntOp>(loc, 16, i32);
+  loadOp = bt0.create<arith::ShLIOp>(loc, loadOp, cst16Op);
+  loadOp = bt0.create<arith::BitcastOp>(loc, bt0.getF32Type(), loadOp);
+
+  return loadOp;
+}
+
+static mlir::Value convertFloatToBF16ToFloat(OpBuilder bt0,
+                                             mlir::Value loadOp) {
+  auto loc = bt0.getUnknownLoc();
+  mlir::Type i16 = bt0.getIntegerType(16);
+  mlir::Type i32 = bt0.getI32Type();
+  loadOp = bt0.create<arith::BitcastOp>(loc, i32, loadOp);
+  // <loadOp> = <loadOp> & FFFF0000
+  auto cst16Op = bt0.create<arith::ConstantIntOp>(loc, -65536, i32);
+  loadOp = bt0.create<arith::AndIOp>(loc, loadOp, cst16Op);
+  loadOp = bt0.create<arith::BitcastOp>(loc, bt0.getF32Type(), loadOp);
+
+  return loadOp;
+}
```

Comment:
Good thoughts. Let me try it. Thanks.

Reply:
For full generality, having `SoftwareBF16` be a pass over `arith` might be the right call. There's really a bunch of design decisions to make here.

Also, if you want to convert a constant float to a bfloat, `APFloat` and its bitcasting methods might be what you're looking for instead of the function you added here.

Let me know how important it is to get this in so I can triage "let's put it in" vs "let's get this right"